plugins/newrelic/src/api/index.test.ts

```diff
@@ -0,0 +1,384 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NewRelicApplication, NewRelicClient } from '.';
+import { DiscoveryApi } from '@backstage/core-plugin-api';
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+import { MockFetchApi, setupRequestMockHandlers } from '@backstage/test-utils';
+
+const mockedDiscoveryApi: DiscoveryApi = {
+  getBaseUrl: async () => 'https://test.test',
+};
+
+beforeEach(() => {
+  jest.resetAllMocks();
+});
+
+describe('NewRelicClient', () => {
+  const server = setupServer();
+  setupRequestMockHandlers(server);
+
+  beforeEach(() => {
+    server.resetHandlers();
+  });
+
+  test.each([
+    ['https://test.test/BASEPATH/apm/api/applications.json', '/BASEPATH'],
+    ['https://test.test/BASEPATH2/apm/api/applications.json', '/BASEPATH2'],
+    ['https://test.testBASEPATH3/apm/api/applications.json', 'BASEPATH3'],
+    ['https://test.test/newrelic/apm/api/applications.json', undefined],
+  ])(
+    'It correctly forms the request url (%p) when proxyPathBase is %p',
+    async (expectedUrl, basePathOverride) => {
+      server.use(
+        rest.get(expectedUrl, (_, res, ctx) =>
+          res(ctx.status(200), ctx.json({ applications: [] })),
+        ),
+      );
+
+      const mockedFetchApi = new MockFetchApi();
+      const fetchSpy = jest.spyOn(mockedFetchApi, 'fetch');
+
+      const client = new NewRelicClient({
+        discoveryApi: mockedDiscoveryApi,
+        fetchApi: mockedFetchApi,
+        proxyPathBase: basePathOverride,
+      });
+      await client.getApplications();
+
+      expect(fetchSpy).toHaveBeenCalledWith(expectedUrl);
```

Comment:
One cool thing about msw is that it'll absolutely refuse (throws an error) to allow requests to unexpected URLs. So you can just put expects inside the `rest.get` handler and like put an `expect.assertions(2)` or whatnot at the top to make sure that they were all exercised - without spying. But I'll leave it up to future tweaks if wanted

Reply:
I will keep that in mind in the future when working with ```msw```. Thank you 👍

If it is okay, for now with this PR I would like to merge as is with the spy.